api/inc/uvisor_exports.h

```diff
@@ -77,11 +77,18 @@
 /** Static Assertion Macro
  *
  * This macro works from both inside and outside function scope.
- *
- * FIXME This is currently not implemented. This issue is tracked at
- * https://github.com/ARMmbed/uvisor/issues/288
- */
-#define UVISOR_STATIC_ASSERT(cond, msg)
+ * The implementations differ due to compilation differences, C++ static_assert
+ * is known from C++11 (__cplusplus > 199711L) while mbed-os compiles with c++98,
+ * and C files use _Static_assert. */
+#ifdef __cplusplus
+#if __cplusplus > 199711L
+#define UVISOR_STATIC_ASSERT(cond, msg) static_assert(cond, #msg)
+#else
+#define UVISOR_STATIC_ASSERT(cond, msg) typedef char STATIC_ASSERT_##msg[(cond)?1:-1]
+#endif
+#else
+#define UVISOR_STATIC_ASSERT(cond, msg) _Static_assert(cond, #msg)
```

Comment:
Please use elseif to eliminate excessive nesting. Also, check that C is new enough for _Static_assert, and otherwise fallback to the typedef-based implementation.